python3Packages.django_2_2: init at 2.2

lsix opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

This introduces Django-2.2, the new LTR version of django. As it only
supports python3, python2 will continue to use django_1_11 as LTR
version as it is still supported until 2020.

We can decide to continue with 1.11 as main version of django for some time, but I prefer to pick the last stable one.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

lsix committed:
python3Packages.django_2_2: init at 2.2.1

This introduces Django-2.2, the new LTR version of django.

For the time being, django-1.11 continues to be LTR in nixpkgs
django-2.2 is introduced to prepare the migration.

dotlambda commented on pkgs/development/python-modules/django/2_2.nix:
```diff
@@ -0,0 +1,43 @@
+  # patch only $out/bin to avoid problems with starter templates (see #3134)
+  postFixup = ''
+    wrapPythonProgramsIn $out/bin "$out $pythonPath"
+  '';
```
I don't think this does what the comment says it does. Instead, it wraps a second time creating e.g. `..django-admin-wrapped-wrapped`.

dotlambda commented on pkgs/development/python-modules/django/2_2.nix:
```diff
@@ -0,0 +1,43 @@
+    sha256 = "06d1x1a4vqzwmv73sixpcf993fj2rxzrl61617hi8387zgj46dbw";
+  };
+
+  patches = stdenv.lib.optionals withGdal [
```
```suggestion
  patches = stdenv.lib.optional withGdal
```

dotlambda commented on pkgs/top-level/python-packages.nix:
```diff
@@ -2347,7 +2347,7 @@ in {
   crayons = callPackage ../development/python-modules/crayons{ };
 
-  django = self.django_1_11;
+  django = if isPy3k then self.django_2_2 else self.django_1_11 ;
```
```suggestion
  django = if isPy3k then self.django_2_2 else self.django_1_11;
```

dotlambda commented on pkgs/top-level/python-packages.nix:
```diff
@@ -2357,6 +2357,10 @@ in {
   };
 
+  django_2_2 = callPackage ../development/python-modules/django/2_2.nix {
+    gdal = self.gdal;
```
Should not be necessary unless we explicitly want `pkgs.gdal`.

lsix commented:
@dotlambda I have addressed your review. Sorry for such a long delay.

For the time being, I keep django-1.11 as main version, so we can introduce 2.2 and later migrate the default version of django used.
